File: 1051. Height Checker/solution.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:

        max_height = 101
        different_heights = 0
        heights_counter = [0]*max_height

        for height in heights:
             heights_counter[height] +=1
        
        h = 0
        for i in range(max_height):
            for j in range(heights_counter[i]):
                if heights[h] != i:
                    different_heights += 1
                h += 1

        return different_heights

File: 1051. Height Checker/Solution.cs
public class Solution {
    public int HeightChecker(int[] heights) {
        
        const int maxHeight = 101;
        int differentHeights = 0;
        var heightsCounter = new int[maxHeight];

        foreach (int height in heights) heightsCounter[height]++;
        
        for (int i = 0, h = 0; i < maxHeight; ++i) {
            for (int j = 0; j < heightsCounter[i]; ++j) {
                if (heights[h++] != i) differentHeights++;
            }
        }

        return differentHeights;
    }
}
